src/Symfony/Component/Console/Helper/QuestionHelper.php

```diff
@@ -512,11 +512,11 @@ private function isTty(): bool
         $inputStream = !$this->inputStream && \defined('STDIN') ? STDIN : $this->inputStream;
```

Comment:
I just discovered that this is enough to fix the warning:
`$inputStream = null === $this->inputStream || (\defined('STDIN') && STDIN === $this->inputStream) ? fopen('php://input', 'r') : $this->inputStream;`

Reply:
Unfortunately, this trades one error for another. This DOES fix the "bytes of buffered data lost during stream conversion" error. But now, as soon as an answer fails validation, instead of asking again, it ALWAYS fails immediately - it falls out of the loop and hits the `throw` at the end https://github.com/symfony/symfony/blob/5a74790bfd4387010e3695ffd7fddc58ebf1b089/src/Symfony/Component/Console/Helper/QuestionHelper.php#L479

Here is a SUPER simple reproduced - https://github.com/weaverryan/console-question-reproducer - I tried the current code on that and your proposed patch. But this PR doesn't fix the issue either - I think this PR sort of fixes it, but then I think we're still hitting #37046.

The only thing that fully works is reverting back to 5.0.8 - you can try all of this pretty quickly with the reproducer.